src/Core/Probers/MultiByte/UTF8Prober.cs

```diff
@@ -107,11 +119,17 @@ public override float GetConfidence(StringBuilder status = null)
         {
             float unlike = 0.99f;
             float confidence;
+            var mbCharRatio = 0.0f;
+            var nonBasciAsciiLen = fullLen - basicAsciiLen;
+            if (nonBasciAsciiLen > 0)
+            {
+                mbCharRatio = (float)mbCharLen / nonBasciAsciiLen;
+            }
 
-            if (numOfMBChar < 6)
+            if (numOfMBChar < 6 && mbCharRatio <= 0.6)
             {
                 for (int i = 0; i < numOfMBChar; i++)
-                    unlike *= ONE_CHAR_PROB;
+                    unlike *= (float)Math.Pow(ONE_CHAR_PROB, numOfMBChar);

```

Comment:
Hello, sorry for the delay, it took a while to understand this change.

This method can be simplified to the following state:
```cs
public override float GetConfidence(StringBuilder status = null)
{
    const float like = 0.99f;

    if (numOfMBChar >= 6)
        return like;

    var mbCharRatio = (float)mbCharLen / (fullLen - basicAsciiLen);
    if (mbCharRatio > 0.6f)
        return like;

    var negative = (float)Math.Pow(ONE_CHAR_PROB, numOfMBChar * numOfMBChar);
    return like * (1f - negative);
}
```

I found a partial explanation in this PR https://github.com/aadsm/jschardet/pull/59

But this particular change is not entirely clear to me. Asked about it here https://github.com/aadsm/jschardet/pull/57#issuecomment-816665501

Also, I can't figure out why it works?

Reply:
For reference：https://github.com/aadsm/jschardet/commit/f45b273c1fbe51f637fba9b91ce644718be8ee93#r47324263

Reply:
If `ONE_CHAR_PROB = 0.45f`, then tests will start to pass too (instead of double pow)

(... I can't understand out why it works)